doc/reference/reference_lua/checks.rst

```diff
@@ -0,0 +1,516 @@
+-   A :ref:`table type qualifier <checks_table_type_qualifiers>` checks whether the values of a table passed as an argument conform to the specified types.
+
+.. NOTE::
+
+    For earlier versions, you can install the ``checks`` module from the `Tarantool rocks repository <https://www.tarantool.io/en/download/rocks>`_.
+
+.. _checks_loading:
+
+Loading checks
+--------------
+
+The :ref:`checks API <checks_api_reference>` is available in a script without loading the module using ``require('checks')``.
+
+
+.. _checks_number_of_arguments:
+
+Number of arguments to check
+----------------------------
+
+Depending on the number of arguments to be checked, you can pass one or more type qualifiers to the :ref:`checks(type_1, ...) <checks-checks>` function.
+
+.. _checks_one_argument:
+
+One argument
+~~~~~~~~~~~~
+
+In the example below, the ``checks`` function accepts a ``string`` type qualifier to verify that only a string value can be passed to the ``greet`` function.
+Otherwise, an error is raised.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+.. _checks_multiple_arguments:
+
+Multiple arguments
+~~~~~~~~~~~~~~~~~~
+
+To check the types of several arguments, you need to pass the corresponding type qualifiers to the ``checks`` function.
+In the example below, both arguments should be string values.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_multiple_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+To turn off checking of specific arguments, use the :ref:`? placeholder <checks_any_type>`.
+
+
+.. _checks_variable_number_of_arguments:
+
+Variable number of arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can check the types of explicitly specified arguments for functions that accept a variable number of arguments.
+
+..  literalinclude:: /code_snippets/test/checks/checks_vararg_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+
+.. _checks_string_type_qualifier:
+
+String type qualifier
+---------------------
+
+This section describes how to check a specific argument type using a string type qualifier:
+
+- The :ref:`Supported types <checks_string_supported_types>` section describes all the types supported by the ``checks`` module.
+- If required, you can make a :ref:`union type <checks_union_type>` to allow an argument to accept several types.
+- You can make any of the supported types :ref:`optional <checks_optional_type>`.
+- To turn off checking of specific arguments, use the :ref:`? placeholder <checks_any_type>`.
+
+.. _checks_string_supported_types:
+
+Supported types
+~~~~~~~~~~~~~~~
+
+.. _checks_lua_types:
+
+Lua types
+*********
+
+A string type qualifier can accept any of the `Lua types <https://www.lua.org/pil/2.html>`_, for example, ``string``, ``number``, ``table``, or ``nil``.
```

Comment:
Why not just "can be any Lua type"? It is easier to read.

Reply:
Strictly speaking, `type qualifier != type`. A type qualifier can accept one type, several types, the `?` value. `Accept` sounds more technically correct here.